comment/static/js/comment.js

```diff
@@ -305,6 +306,137 @@ $(function() {
         });
     };
 
+    /**
+     * Create a temporary div, append it to the div'#response', fix it to the top and fade it.
+     * @param {string} responseEle - The DOM element to be used for displaying in the response.
+     * @param {int} status - an integer based upon the response received for AJAX request.
+     * (-1->'error'|0->'success'| 1->'warning')
+     * @param {string} msg - a string depicting the message to be displayed in the response. 
+     * @param {int} time - time after which the response fades away
+     */
+    function createResponse(responseEle, status, msg, time = 5000) {
+        switch (status) {
+            case -1:
+                status = "danger";
+                break;
+            case 0:
+                status = "success";
+                break;
+            case 1:
+                status = "warning";
+        }
+        var cls = 'h6 alert alert-' + status;
+        var temp = $('<div/>')
+            .addClass(cls)
+            .html(msg);
+        responseEle.prepend(temp[0]);
+        fixToTop(temp);
+        temp.fadeIn(time);
+        temp.fadeOut(2 * time);
+        setTimeout(function() {
+            temp.remove();
+        }, 2 * time + 10);
+    }
+    /**
+     * Fixes an element to the top of the viewport.
+     * @param {object} div - element that is to be fixed at the top of the viewport. 
+     */
+    function fixToTop(div) {
+        var top = 200;
+        var isfixed = div.css('position') == 'fixed';
+        if (div.scrollTop() > top && !isfixed)
+            div.css({ 'position': 'fixed', 'top': '0px' });
+        if (div.scrollTop < top && isfixed)
+            div.css({ 'position': 'static', 'top': '0px' });
+    }
+
+    /**
+     * Perform an AJAX request to the flag API and handle response.
+     * @param {object} $flag - the parent DOM element that contains flag. 
+     * @param {string} action - 'create'/'delete'.
+     * @param {string} reason - the value of the reason for flagging, default=null.
+     * @param {string} info - any extra information to be passed when flagging, default=null.
+     */
+    var sendFlag = function($flag, action, reason = null, info = null) {
```

Comment:
the `action` argument is no longer required here.

Reply:
I didn't look at JS code at all. This will be fixed.